com.unity.render-pipelines.high-definition/Editor/VFXGraph/Shaders/VFXDecal.template

```diff
@@ -33,8 +33,7 @@ if(_EnableDecalLayers)
 	float3 clipPos;
 	clipPos.xy = (i.pos.xy / _ScreenParams.xy) * 2.0f - 1.0f;
 	clipPos.z = VFXSampleDepth(i.pos);
-	if (_ProjectionParams.x < 0)
```

Comment:
Just had a second thought about that, and I believe this shader code is correct, and that the issue comes from URP not setting projectionParams.x properly, and/or not setting UNITY_MATRIX_I_VP properly.
I'm confused about where these are defined exactly, because they seem to be set from C# first (in ScriptableRenderer.cs l.210, and in RenderingUtils.cs, l.138), and later from C++ (GfxDevice.cpp).

Reply:
Mhh, you mean the _ProjectionParams.x should not have been inverted in overlay.
I made a renderdoc capture and indeed, the only difference is about `_ProjectionParams.x`
[_Base_CS.csv](https://github.com/Unity-Technologies/Graphics/files/7032430/_Base_CS.csv)
[_Overlay_CS.csv](https://github.com/Unity-Technologies/Graphics/files/7032431/_Overlay_CS.csv)
![image](https://user-images.githubusercontent.com/22494706/130458910-27aacb4f-341b-42ab-b6e0-9957d772a360.png)

So, maybe we can check if commenting this `if` : https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Runtime/RenderingUtils.cs#L195 changes something ⏳

Reply:
Yes, when looking at a Renderdoc capture, both Base Camera and Overlay use the flipped texture, but Overlay has _ProjectionParams.x to +1.

When debugging at this line https://github.com/Unity-Technologies/Graphics/blob/70f78edee0ed1e28d62a6584a59bcee5822ccb6d/com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs#L211, `projectionFlipSign` is set to +1 even for the Base Camera, but subsequently (in C++ from what I understand), it gets set to -1.

Reply:
The inverse matrix seems correctly computed. I'm wondering if it isn't a wrong consideration here : https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs#L209
**Before**
`bool isOffscreen = cameraData.targetTexture != null;`
**After**
`bool isOffscreen = cameraData.targetTexture != null || cameraData.renderType == CameraRenderType.Overlay;`

Reply:
I agree it doesn't feel right..

Reply:
Following up in this [conversation](https://unity.slack.com/archives/GBJ7AU4V8/p1629731647074000)

Reply:
Hi I don't have access to the conversation but to me projectionparams is needed. I would say it should be directly `clipPos.y *= _ProjectionParams.x`
If there's a issue with the overlay mode not setting the value correctly, this should be fixed in that code as part of this PR

Reply:
I opened an independant fogbugz about the overlay issue: https://fogbugz.unity3d.com/f/cases/1368762/